### ISSUE TITLE
Certificate revocation signed from account key

### ISSUE THREAD

**Issue by aloopkin** [shred/acme4j#73]
Hello,

I can't find how to revoke a certificate using a message signed by the account key (and not by the certificate associated private key).
This behaviour is allowed by the RFC (speaking as of draft 18), and there's indeed a revoke() method in the Certificate object. Unfortunately, this implies that you have already a Certificate object, and it is returned following to enrollment: i don't see how to rebuild it later. Maybe i missed something?
Another alternative would be to add a static method similar to the other Certificate.revoke() method, that would do just that.

Regards,
Alexandre

**Comment by aloopkin** [comment]
for the record, i think this code should work with a bit of refactoring:
```
public static void revoke(Session session, X509Certificate certificate, Account acct, RevocationReason reason) {
		try {
			URL resUrl = session.resourceUrl(Resource.REVOKE_CERT);
			if (resUrl == null) {
				throw new AcmeException("Server does not allow certificate revocation");
			}
			Connection conn = session.connect();
			JSONBuilder claims = new JSONBuilder();
			claims.putBase64("certificate", certificate.getEncoded());
			if (reason != null) {
				claims.put("reason", reason.getReasonCode());
			}
			conn.sendSignedRequest(resUrl, claims, acct.getLogin());
		} catch (CertificateEncodingException ex) {
			throw new AcmeException("Could not revoke certificate", ex);
		} 
}
```

**Comment by shred** [comment]
If you know the location URL of the certificate, you can do this:

```
URL certificateUrl = // Location URL of your certificate

Certificate cert = login.bindCertificate(certificateUrl);
cert.revoke(reason);
```

You can get the location URL of your certificate via `certificate.getLocation()` after creating it.

As an alternative, you can revoke a certificate with the certificate itself, and the corresponding domain key pair:

```
KeyPair domainKeyPair = // Key Pair that is used for HTTPS encryption
X509Certificate certificate = // Matching certificate to revoke

Certificate.revoke(session, domainKeyPair, certificate, reason);
```

Instead of the `domainKeyPair`, you can also use your account key pair here. It is undocumented, but should work. I guess this is what you are looking for.

I will add this use case to the documentation of the static `Certificate.revoke()` method.

**Comment by aloopkin** [comment]
I don't store the location URL of the certificate, and i don't have access to its private key (trying to cover the "certificate private key loss" use case), so it doesn't work for me unfortunately.

Using the existing static method with the account key doesn't work either, because in the case of the account key, the signature should contain the kid, not the JWK (see #62 , still true as of draft 18).

Hence the method i suggested...

**Comment by shred** [comment]
Yes, you're right... I forgot about the JWK/kid problem. I will add another method.

**Comment by aloopkin** [comment]
Danke sehr!
